src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+
+#### Example: Aligning text
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example displays "Hey!" three times in one row,
+  but at different font sizes and in different colors.
+  `Row` specifies the `crossAxisAlignment` and `textBaseline` properties.
+
+  As mentioned previously, to align `Text` widgets along their character baselines,
+  set the `textBaseline` property to `TextBaseline.alphabetic`.  
+
+  **1.** Click the **Run** button. 
+
+  **2.** Change `CrossAxisAlignment.center` to `CrossAxisAlignment.baseline`, and run again.
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/0ff109090b99ef1873d9fad501b2bc86
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=0ff109090b99ef1873d9fad501b2bc86&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Icon widget
+
+The `Icon` widget displays a graphical symbol, 
+which represents an aspect of the UI. 
+Flutter is preloaded with icon packages for 
+[Material](https://api.flutter.dev/flutter/material/MaterialApp-class.html) and 
+[Cupertino](https://api.flutter.dev/flutter/cupertino/CupertinoApp-class.html) applications.
+
+#### Example: Create an Icon widget
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example displays the widget icon from the
+  [Material Icon library](https://api.flutter.dev/flutter/material/Icons-class.html) in red and blue.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Add another `Icon` widget from the 
+  [Material Icon library](https://api.flutter.dev/flutter/material/Icons-class.html)
+  with a size of 50. 
+
+  **3.** Give the icon a color of `Color.amber` from the 
+  [Material Color palette](https://api.flutter.dev/flutter/material/Colors-class.html), 
+  and run again.  
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/01688fca8c13f85d93078054af2e858b
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=01688fca8c13f85d93078054af2e858b&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Image widget 
+
+The `Image` widget displays an image. You either can reference images using a URL,
+or you can include images inside your app package. Since DartPad can't package an image,
+the following example uses an image from the network.
+
+#### Example: Image widget
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example displays an image using `Image`. 
+  The `Image.network` method takes a string parameter that contains the image's URL.
+
+  This example displays an image that's stored remotely on [GitHub](https://github.com/flutter/website/tree/master/examples/layout/sizing/images).
+
+  **1.** Click the **Run** button.  
+{{site.alert.end}}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=b6f3084800bd139cdb522b8858bb58b7&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe>
+
+## Putting it all together 
+
```

Comment:
It seems weird to have a layout section that has no figures at all. I expected to see the final UI, then break it down into Rows & Columns. These you'd then break into the parts.